src/shipchain_common/rpc.py

```diff
@@ -45,25 +45,38 @@ def call(self, method, args=None):
 
         try:
             with TimingMetric('engine_rpc.call', tags={'method': method}) as timer:
-                response_json = settings.REQUESTS_SESSION.post(self.url,
-                                                               data=json.dumps(self.payload, cls=DecimalEncoder),
-                                                               timeout=getattr(settings, 'REQUESTS_TIMEOUT', 270)
-                                                               ).json()
+                response = settings.REQUESTS_SESSION.post(self.url, data=json.dumps(self.payload, cls=DecimalEncoder),
+                                                          timeout=getattr(settings, 'REQUESTS_TIMEOUT', 270))
+                response_json = response.json()
+
                 LOG.info('rpc_client(%s) duration: %.3f', method, timer.elapsed)
 
-            if 'error' in response_json:
+            if 'error' in response_json and response.status_code == status.HTTP_200_OK:
+                # It's an error properly handled by engine
                 log_metric('engine_rpc.error', tags={'method': method, 'code': response_json['error']['code'],
                                                      'module': __name__})
                 LOG.error('rpc_client(%s) error: %s', method, response_json['error'])
                 raise RPCError(response_json['error']['message'])
 
+            if status.HTTP_400_BAD_REQUEST <= response.status_code:
+                # It's an unexpected error not handled by engine
+                log_metric('engine_rpc.error', tags={'method': method, 'code': response.status_code,
+                                                     'module': __name__})
+                LOG.error('rpc_client(%s) error: %s', method, response_json)
+                raise RPCError(str(response_json), status_code=response.status_code)
```

Comment:
Casting a dict to a string is not going to be as readable as `json.dumps(response_json)`. I think that'll give us nicer error messages.

Also, I'm not sure how I feel about passing back any non-500 level error messages from RPC calls. I'm thinking almost any of the errors that could occur here should be considered a HTTP_500_INTERNAL_SERVER_ERROR. @mlclay thoughts?

Reply:
This was essentially for debugging purposes that engine's response status code is passed in here since it refers to edge cases for which the request is unsuccessful and not handled  by engine. Brief status codes different than 200.

Reply:
I think any error generated during an RPC call should be returned to the user as a 500 level error.  Consumers of the RPC client (i.e. Transmission) are responsible for forming the parameters to Engine RPC, not the actual end-users of the Transmission API.  A 400 level error would indicate they could correct the problem by modifying their parameters, but that is not likely to be the case.

Regarding the discussion on re-raising the RPCError: If we in fact no longer care about the status code of the raised RPCErrors, then the we may no longer need to re-raise the RPCError exception (which in my understanding is only to preserve the status codes) as the string representation of the error should suffice.